codegenerator/cli/npm/envio/src/Debouncer.res

```diff
@@ -0,0 +1,65 @@
+type t = {
+  mutable lastRunTimeMillis: float,
+  mutable isRunning: bool,
+  mutable isAwaitingInterval: bool,
+  mutable scheduled: option<unit => promise<unit>>,
+  intervalMillis: float,
+  logger: Pino.t,
+}
+
+let make = (~intervalMillis: int, ~logger) => {
+  lastRunTimeMillis: 0.,
+  isRunning: false,
+  isAwaitingInterval: false,
+  scheduled: None,
+  intervalMillis: intervalMillis->Belt.Int.toFloat,
+  logger,
+}
+
+let t = (logger, exn, message) =>
+  logger->Pino.errorExn(message->Pino.createPinoMessageWithError(exn))
+
+%%private(
```

Comment:
Let's have a .resi file instead

Reply:
Can you elaborate on why you think so? 

IMO this is better for the given case unless we want opaque types. Since you don't need to maintain 2 files and keep signatures/types matching.

Reply:
Ok, I thought we want it to be opaque

Reply:
I like opaque types from an API perspective, but it makes it less testable and more encapsulated. I'm keen to try not use too much encapsulation unless it solves footguns. At which point you can just have an inner module of exposed getter functions for testing.

Reply:
I prefer the black-box approach for unit testing, when we test only the public API the way how the user would interact with it. This is more reliable since we test exactly how the code is used. Also, when some internal implementation changes, it doesn't require us to update the tests.

Reply:
Fair, but it's still more boilerplate just to hide simple data etc. I'm wondering how much the black-box ceremony helps us vs just creates overhead.

Reply:
I got used to writing the code this way, so I think this is not a big difference. But might be annoying sometimes